async-riscv/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+//! Proof of Concept async runtime for the Cortex-M architecture
+
+#![deny(missing_docs)]
+#![deny(rust_2018_idioms)]
+#![deny(warnings)]
+#![no_std]
+
+mod alloc;
+mod executor;
+pub mod task;
+pub mod unsync;
+
+/// This keeps dropping into the debugger and never returns
+pub fn abort() -> ! {
+    loop {
+        unsafe { riscv::asm::ebreak() }
+    }
+}
+
+#[cfg(feature = "wait-nop")]
```

Comment:
@mwkroening @japaric I can move ARM implementations here, too. The rest of the crate is generic and works both for ARM and for RISC-V.

Reply:
I like where this goes, but I have no authority here, so someone else would need to review this.